File: tang-commons/src/main/java/com/tang/commons/enumeration/LoginType.java
package com.tang.commons.enumeration;

import org.apache.commons.lang3.StringUtils;

/**
 * 登陆方式
 *
 * @author Tang
 */
public enum LoginType {

    /**
     * 用户名密码
     */
    USERNAME("username", "用户名密码"),

    /**
     * 邮箱密码
     */
    EMAIL("email", "邮箱密码"),

    /**
     * GitHub 授权码
     */
    GITHUB("github", "GitHub 授权码"),

    /**
     * 未知
     */
    UNKNOWN("", "未知");

    /**
     * 登陆方式
     */
    private final String name;

    /**
     * 登陆方式名称
     */
    private final String label;

    LoginType(String name, String label) {
        this.name = name;
        this.label = label;
    }

    public String getName() {
        return name;
    }

    public String getLabel() {
        return label;
    }

    /**
     * 根据 name 返回枚举类型
     *
     * @param name 登陆方式
     * @return {@link LoginType} 登陆方式名称
     */
    public static LoginType getLoginType(String name) {
        if (StringUtils.isBlank(name)) {
            return UNKNOWN;
        }
        for (LoginType loginType : LoginType.values()) {
            if (loginType.getName().equals(name)) {
                return loginType;
            }
        }
        return UNKNOWN;
    }

}

File: tang-commons/src/main/kotlin/com/tang/commons/enumeration/LoginType.kt
package com.tang.commons.enumeration

/**
 * 登陆方式
 *
 * @author Tang
 */
enum class LoginType(val value: String, val label: String) {

    /**
     * 用户名密码
     */
    USERNAME("username", "用户名密码"),

    /**
     * 邮箱密码
     */
    EMAIL("email", "邮箱密码"),

    /**
     * GitHub 授权码
     */
    GITHUB("github", "GitHub 授权码"),

    /**
     * 未知
     */
    UNKNOWN("", "未知");

    companion object {

        /**
         * 根据 value 返回枚举类型
         *
         * @param value 登陆方式
         * @return {@link LoginType} 登陆方式名称
         */
        @JvmStatic
        fun getLoginType(value: String): LoginType {
            return entries.find { it.value == value } ?: UNKNOWN
        }

    }

}
